androidApp/src/main/java/com/EENX15_22_17/digital_journal/android/MainActivity.kt

```diff
@@ -6,15 +6,19 @@ import androidx.activity.compose.setContent
 import androidx.compose.material.Text
 import androidx.compose.material.Button
 import androidx.compose.runtime.Composable
+import androidx.lifecycle.ViewModelProvider
+import com.EENX15_22_17.digital_journal.android.screens.arrival.patientinfo.ArrivalViewModel
 import com.EENX15_22_17.digital_journal.android.ui.theme.DigitalJournalTheme
 
 
 class MainActivity : ComponentActivity() {
+    private lateinit var arrivalViewModel: ArrivalViewModel
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        arrivalViewModel = ViewModelProvider(this)[ArrivalViewModel::class.java]
         setContent {
             DigitalJournalTheme {
-                NavigationApp()
+                NavigationApp(arrivalViewModel = arrivalViewModel)
             }
```

Comment:
Well done!

I suppose that it will probably be more  parameters that need to be sent into NavigationApp. The remaining ViewModels like:   NavigationApp(arrivalViewMode = arrivalVieWModel, suicideAssessmentViewModel = suicdeViewModel, ...)

It's fine and works well 

But maybe in the aim of getting more readable code (Parameter Object Pattern), we can consider to create a dataclass that will represent the group of parameters like this:

data class ViewModelData(
    val arrivalViewModel : ArrivalViewModel,
    val suicdeViewModel: SuicideAssessmentViewModel,
    ....
)
private lateinit var arrivalViewModel: ArrivalViewModel
arrivalViewModel = ViewModelProvider(this)[ArrivalViewModel::class.java]

private lateinit var suicideViewModel: SuicideAssessmentViewModel
suicideViewModel= ViewModelProvider(this)[SuicideAssessmentViewModel::class.java]

viewModelData = ViewModelData(
          arrivalViewModel  = arrivalViewModel,
          suicdeViewModel = suicideViewModel
 )

And then:

NavigationApp(viewModelData)

fun ArrivalPage(
    arrivalViewModel: ViewModelData.arrivalViewModel,
    visitId: String,
    navBack: () -> Unit
) { .. }

And so on...